### ISSUE TITLE
`curly` and comments

### ISSUE THREAD

**Issue by wwwillchen** [eslint/eslint#7538]
**Tell us about your environment**

* **ESLint Version:** 3.9.1
* **Node Version:** 4.5.0
* **npm Version:** 2.15.11

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

```
"curly": [2, "multi-or-nest", "consistent"]
```

**What did you do? Please include the actual source code causing the issue.**

```js
if (foo) {
  // some comment
  bar();
}
```

**What did you expect to happen?**
ESLint should not give an error for having a brace because the block is multi-line, however ESLint is treating it as a single-line statement and does not observe the comment.

**What actually happened? Please include the actual, raw output from ESLint.**
`Unnecessary { after 'if' condition.`

ESLint essentially wants no braces which is confusing to myself (and likely other JS readers):

```
if (foo)
  // some comment
  bar();
```

---

I'm happy to work on this bug as my project at work is interested in using this rule. I'm thinking it will be a fairly simple to fix, along the lines of https://github.com/eslint/eslint/pull/6396

**Comment by not-an-aardvark** [comment]
Thanks for the report; this does seem like a bug.

**Comment by mysticatea** [comment]
It seems a bug.

I felt the name `multi-or-nest` is confusing since the rule has the option `multi` and `multi-line` then the `multi-or-nest`'s behavior is near to `multi-line`.

**Comment by wwwillchen** [comment]
I agree the naming is confusing and I had to re-read the examples a few times to fully understand the differences.  If it helps, the reason why our project is using multi-or-nest is because through some (strange-ish) transitive logic, we follow a variation of the Google C++ style guide [0] which specifies that "In general, curly braces are not required for single-line statements". I'm not sure what a better name is, perhaps "multi-line-statement"?

[0] https://google.github.io/styleguide/cppguide.html#Conditionals
